feature: Release notes for scheduled DB maintenance CY-3325 (#716)
codacy/docs: docs/release-notes/cloud/scheduled-db-maintenance.md

This operation will include around 2 hours of initial downtime, followed by 8 hours of degraded analysis performance for all users and an additional 5 hours for open source users. We expect to finish the maintenance operation at around July 3, 2021 23:00 UTC.

Our [status page](https://status.codacy.com/) will always include the most up-to-date information regarding the operational status of Codacy Cloud.

## How does this impact me?

To mitigate the impact, as part of the maintenance operation we'll re-analyze the pull requests and the last commit in the default branch of each repository according to the following retention periods:

-   6 months for **Pro and Trial plans**
-   15 days for **Open Source plans**

The following is the expected impact during the maintenance operation:

-   New commits and pull requests will either not be picked up by Codacy or take longer to analyze. If you have set Codacy as a required check to merge it will fail.

The following is the expected impact after the maintenance operation is complete:

> -   Historical static code analysis information will be missing for all existing commits and pull requests, except for the ones that we re-analyze as part of the maintenance operation

Author: I believe so, as the part that we've been adjusting is this one:

https://github.com/codacy/docs/blame/5debc9d4712b930d9c7e633f6891b39f32c0f51b/docs/release-notes/cloud/scheduled-db-maintenance.md#L18

Perhaps a more straightforward way of conveying this information could be:

```suggestion
-   Historical static code analysis information will be missing for any commits and pull requests that we don't re-analyze as part of the maintenance operation
```

What do you think?
